docs/hsm.md

```diff
@@ -0,0 +1,90 @@
+# Dinamo Networks HSM integration specs
+
+## Description
+
+The HSM integration allows you to connect your wallet with the HSM of Dinamo Networks.
+This means the wallet is managed by the headless but the private keys are stored in the HSM.
+
+## Configuration
+
+The configuration requires the HSM host, user and password to connect to the HSM.
+
+You will also need to create an HSM key to use with the wallet.
+
+### Docker
+
+Environment variables:
+
+- `HEADLESS_HSM_HOST`: HSM host
+- `HEADLESS_HSM_USERNAME`: HSM username
+- `HEADLESS_HSM_PASSWORD`: HSM password
+
+CLI arguments:
+
+- `--hsm-host`: HSM host
+- `--hsm-username`: HSM username
+- `--hsm-password`: HSM password
+
+### Running locally
+
+When running locally you write the `src/config.js` file and set the configuration in it.
+The config names are slightly different but work the same way as in the docker config.
+
+- `hsmHost`
+- `hsmUsername`
+- `hsmPassword`
+
+### Creating an HSM key
+
+You will need to run a command to create the HSM key in the docker container
+
+The following command will create a key named `abc`.
+
+```bash
+docker run --env-file=./.env --entrypoint make hathornetwork/hathor-wallet-headless create_hsm_key keyname=abc
+```
+
+The `.env` file contains the configuration needed to connect to the HSM.
```

Comment:
suggestion: This part felt slightly confusing to me. Maybe just by linking to [our official docs](https://docs.hathor.network/how-to-guides/headless-wallet/install-docker-container) would help clarify this part here.

Reply:
I don't think this is related to running the headless with docker. It's just saying that we need the env vars of HSM configuration in this file, the vars that were described above.